src/js/legend_control.js

```diff
@@ -0,0 +1,651 @@
+   *
+   * @param rows An array of table rows containing placeholders for descriptions.
+   */
+  async prettifyNetworkLabels(rows) {
+    let networkMetadata = await this.getNetworkMetadata();
+    if (!networkMetadata) return;
+
+    for (let row of rows) {
+      let network = row.dataset.network;
+      if (network?.startsWith("omt-gb-")) {
+        let ukNetworkMetadata = await this.getUKNetworkMetadata();
+        Object.assign(networkMetadata, ukNetworkMetadata);
+        break;
+      }
+    }
+
+    let toSentenceCase = (lowerCase, locale) =>
+      lowerCase[0].toLocaleUpperCase(locale) + lowerCase.substring(1);
+    for (let row of rows) {
+      let network = row.dataset.network;
+      let binding = networkMetadata[network];
+      if (!binding) continue;
+
+      let locale = binding?.networkLabel["xml:lang"];
+      let descriptionCell = row.querySelector(".description");
+      if (locale) {
+        descriptionCell.setAttribute("lang", locale);
+      }
+      let label = binding.networkLabel.value;
+      descriptionCell.textContent = toSentenceCase(label, locale);
+    }
+  }
+
+  /**
+   * Returns a mapping from `network=*` values to metadata about these values from Wikidata.
+   */
+  async getNetworkMetadata() {
+    if (this._networkMetadata) {
+      return this._networkMetadata;
+    }
+
+    let url = `https://query.wikidata.org/sparql?query=${encodeURIComponent(
+      this.getNetworkMetadataQuery()
+    )}&format=json`;
+    const response = await fetch(url);
+    const json = await response.json();
+    this._networkMetadata = Object.fromEntries(
+      json.results.bindings.map((binding) => {
+        return [binding.value.value, binding];
+      })
+    );
+    return this._networkMetadata;
+  }
+
+  /**
+   * Returns a mapping from OpenMapTiles synthesized `network=*` values to
+   * metadata about these values from Wikidata.
+   */
+  async getUKNetworkMetadata() {
+    if (this._ukNetworkMetadata) {
+      return this._ukNetworkMetadata;
+    }
+
+    let url = `https://query.wikidata.org/sparql?query=${encodeURIComponent(
+      this.getNetworkMetadataQuery("GB")
+    )}&format=json`;
+    const response = await fetch(url);
+    const json = await response.json();
+    this._ukNetworkMetadata = Object.fromEntries(
+      json.results.bindings.map((binding) => {
+        return [binding.value.value, binding];
+      })
+    );
+    return this._ukNetworkMetadata;
+  }
+
+  purgeNetworkMetadata() {
+    delete this._networkMetadata;
+    delete this._ukNetworkMetadata;
+  }
+
+  /**
+   * Returns the Wikidata Query Service SPARQL query for network metadata.
+   *
+   * @param region ISO 3166-1 alpha-2 country code.
+   */
+  getNetworkMetadataQuery(region) {
+    let locales = Label.getLocales().join(",");
+    let triple,
+      filter = "",
+      bind = "";
+    if (region === "GB") {
+      triple =
+        "?network wdt:P361 wd:Q115856945; p:P528 [ ps:P528 ?value; pq:P972 wd:Q110613756 ]";
```

Comment:
In support of #633, this is an extra query for [parts of](https://www.wikidata.org/wiki/Property:P361) the [UK road coloring scheme](https://www.wikidata.org/wiki/Q115856945) that’s requested whenever the synthesized UK “networks” are listed in the table.